x-pack/plugins/fleet/server/services/agents/action_status.ts

```diff
@@ -4,60 +4,78 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-
 import type { ElasticsearchClient } from '@kbn/core/server';
-import pMap from 'p-map';
 
 import { SO_SEARCH_LIMIT } from '../../constants';
 
-import type { FleetServerAgentAction, ActionStatus } from '../../types';
+import type { FleetServerAgentAction, ActionStatus, ListWithKuery } from '../../types';
 import { AGENT_ACTIONS_INDEX, AGENT_ACTIONS_RESULTS_INDEX } from '../../../common';
+import { appContextService } from '..';
 
 /**
  * Return current bulk actions
  */
-export async function getActionStatuses(esClient: ElasticsearchClient): Promise<ActionStatus[]> {
-  let actions = await _getActions(esClient);
-  const cancelledActionIds = await _getCancelledActionId(esClient);
+export async function getActionStatuses(
+  esClient: ElasticsearchClient,
+  options: ListWithKuery
+): Promise<ActionStatus[]> {
+  const actions = await _getActions(esClient, options);
+  const cancelledActions = await _getCancelledActions(esClient);
+  let acks: any;
 
-  // Fetch acknowledged result for every action
-  actions = await pMap(
-    actions,
-    async (action) => {
-      const { count } = await esClient.count({
-        index: AGENT_ACTIONS_RESULTS_INDEX,
-        ignore_unavailable: true,
-        query: {
-          bool: {
-            must: [
-              {
-                term: {
-                  action_id: action.actionId,
-                },
-              },
-            ],
+  try {
+    acks = await esClient.search({
+      index: AGENT_ACTIONS_RESULTS_INDEX,
+      query: {
+        bool: {
+          // There's some perf/caching advantages to using filter over must
+          // See https://www.elastic.co/guide/en/elasticsearch/reference/current/query-filter-context.html#filter-context
```

Comment:
Appreciate the reference here - not a very obvious choice to use filter here but the link makes it clear.